packages/airnode-node/src/evm/templates/template-fetching.ts

```diff
@@ -80,13 +83,43 @@ export async function fetch(
   apiCalls: Request<ApiCall>[],
   fetchOptions: FetchOptions
 ): Promise<LogsData<ApiCallTemplatesById>> {
-  const templateIds = apiCalls.filter((a) => a.templateId).map((a) => a.templateId);
+  const templateIds = apiCalls.reduce((acc: string[], apiCall) => {
+    if (apiCall.templateId) return [...acc, apiCall.templateId];
+    return acc;
+  }, []);
+
   if (isEmpty(templateIds)) {
     return [[], {}];
   }
 
+  // Check if templateIds are found in config.json
+  const configTemplates = fetchOptions.configTemplates ?? {};
+  const validTemplatesInConfig = templateIds.reduce((acc: ApiCallTemplatesById, id) => {
+    const configTemplateMatch = configTemplates[id];
+
+    if (configTemplateMatch) {
+      // Verify templateIds
+      const expectedTemplateIdV0 = getExpectedTemplateIdV0({
+        airnodeAddress: fetchOptions.airnodeRrpAddress,
+        endpointId: configTemplateMatch.endpointId,
+        encodedParameters: configTemplateMatch.encodedParameters,
+      });
```

Comment:
This seems like something we must verify during config validation with a `refine` or `superRefine` function (packages/airnode-validator/src/config/config.ts)

Reply:
Good idea. I will need to add **airnodeAddress** as an additional field as it's needed  to validate V0 templates :+1:

Reply:
You make a good point. `airnodeAddress` is not currently in config.json 🤔 We do have access to the airnode wallet mnemonic tho